modules/3d-tiles/src/lib/filesystems/tiles-3d-archive-file-system.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  FileProvider,
+  ZipFileSystem,
+  cdSignature as cdHeaderSignature,
+  searchFromTheEnd,
+  parseZipCDFileHeader,
+  HashElement,
+  parseHashFile,
+  parseZipLocalFileHeader
+} from '@loaders.gl/zip';
+import {TZ3Archive} from '../../tz3/tz3-archive';
+
+/**
+ * FileSystem adapter for a 3tz (3D tiles archive format) file
+ * Holds FileProvider object that provides random access to archived files
+ */
+export class Tiles3DArchiveFileSystem extends ZipFileSystem {
+  hashData?: HashElement[] | null;
+
+  /**
+   * Constructor
+   * @param file - instance of FileProvider or file path string
+   */
+  constructor(file: FileProvider | string) {
+    super(file);
+  }
+
+  /**
+   * Implementation of fetch against this file system
+   * @param filename - name of a file
+   * @returns - Response with file data
+   */
+  async fetch(filename: string): Promise<Response> {
+    const fileProvider = await this.fileProvider;
+    if (!fileProvider) {
+      throw new Error('No data detected in the zip archive');
+    }
+    await this.parseHashFile();
+    if (this.hashData) {
+      const archive = new TZ3Archive(fileProvider, this.hashData);
```

Comment:
This looks odd. We are creating one archive on every fetch? Are there multiple archives in here?

Reply:
It is the same. We create it to use `getFile` method